### DISCUSSION TITLE
🔥🔥 New Release 0.51.0 🔥🔥

### DISCUSSION THREAD

**Discussion by theCyberTech** [crewAIInc/crewAI#1168] [Announcements]
- crewAI Testing / Evaluation - https://docs.crewai.com/core-concepts/Testing/
- Adding new sliding context window
- Allowing all attributes on YAML - https://docs.crewai.com/getting-started/Start-a-New-CrewAI-Project-Template-Method/#customizing-your-project
- Adding initial Pipeline Structure - https://docs.crewai.com/core-concepts/Pipeline/
- Ability to set LLM for planning step - https://docs.crewai.com/core-concepts/Planning/
- New crew run command - https://docs.crewai.com/getting-started/Start-a-New-CrewAI-Project-Template-Method/#running-your-project
- Saving file now dumps dict into JSON - https://docs.crewai.com/core-concepts/Tasks/#creating-directories-when-saving-files
- Using verbose settings for tool outputs
- Added new Github Templates
- New Vision tool - https://docs.crewai.com/tools/VisionTool/
- New DALL-E Tool - https://docs.crewai.com/tools/DALL-ETool/
- New MySQL tool - https://docs.crewai.com/tools/MySQLTool/
- New NL2SQL Tool - https://docs.crewai.com/tools/NL2SQLTool.md
- Bug Fixes:
    - Bug with planning feature output
    - Async tasks for hierarchical process
    - Better pydantic output for non OAI models
    - JSON truncation issues
    - Fix logging types
    - Only import AgentOps if the Env Key is set
    - Sanitize agent roles to ensure valid directory names (Windows)
    - Tools name shouldn't contain space for OpenAI
    - A bunch of minor issues

**Comment by Jsin01** [comment]
Need examples of whats supported in the yaml file and the format that is expected. For example, Allowing all attributes on YAML: how do i add the tool at the agent level? tools: [search_tool, web_rag_tool] ? 

Also, do some parts of the config stay in the crew.py and others move to the yaml? 

FYI the CLI does not have anything in the agents.yaml except the same role, goal, backstory. Would be nice to see examples commented out in there.

**Comment by alanwunsche** [comment]
Readme.md could be updated to reflect new crew creation process and  .yaml configuration files.

**Comment by cuongpham-1001** [comment]
@theCyberTech : can you guy take a look on this feature request? https://github.com/crewAIInc/crewAI/discussions/2411
This will be really helpful for us.
